src/main/java/seedu/pill/util/DateTime.java

```diff
@@ -0,0 +1,67 @@
+package seedu.pill.util;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class DateTime {
+    private LocalDateTime dateTime;
+
+    public DateTime() {
+        this.dateTime = LocalDateTime.now();
+    }
+
+    public DateTime(LocalDateTime dateTime) {
+        this.dateTime = dateTime;
+    }
+
+    public LocalDateTime getDateTime() {
+        return dateTime;
+    }
+
+    public void setDateTime(LocalDateTime dateTime) {
+        this.dateTime = dateTime;
+    }
+
+    public String getFormattedDateTime(String format) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
+        return dateTime.format(formatter);
+    }
+
+    public String getFormattedDate() {
+        return getFormattedDateTime("yyyy-MM-dd");
+    }
+
+    public String getFormattedTime() {
+        return getFormattedDateTime("HH:mm:ss");
+    }
+
+    public boolean isAfter(DateTime other) {
+        return this.dateTime.isAfter(other.getDateTime());
+    }
+
+    public boolean isBefore(DateTime other) {
+        return this.dateTime.isBefore(other.getDateTime());
+    }
+
```

Comment:
Should DateTime implement Comparable<DateTime>, such that we can replace isAfter and isBefore with an overriding compareTo method? This would allow for data structures such as TreeSet to leverage this sorting automatically, as they sort via a compareTo method. Just a possibility

Reply:
Oh that's a great idea actually. Thanks, I'll make some changes now.

Reply:
> Should DateTime implement Comparable, such that we can replace isAfter and isBefore with an overriding compareTo method? This would allow for data structures such as TreeSet to leverage this sorting automatically, as they sort via a compareTo method. Just a possibility

Great suggestion. Makes our code more modular too. Truly a 2030 moment.